src/DankCord/core.py

```diff
@@ -86,74 +89,33 @@ def _run_command(self, name: str, retry_attempts= 3, timeout:int = 10):
                     ),
                 )
             )
-            post_handling = self._post_handling(timeout, response, name, nonce, ["MESSAGE_CREATE"])
-            if post_handling:
-                return post_handling
-            continue
-
-    def _post_handling(
-        self,
-        timeout: int,
-        response: Response,
-        name: str,
-        nonce: str,
-        event_type: list,
-        message_id: str = "",
-    ) -> Optional[Message]:
-        _message = None
-
-        if isinstance(response.data, dict):
-            retry_after = int(response.data.get("retry_after", 0))
-            errors = response.data.get("errors")
-            code = response.data.get("code")
-            if errors:
-                # TODO horrible way, sometimes the error does NOT have data, so this needs fixing
-                if response.data["errors"]["data"]["_errors"][0]["code"] == "COMPONENT_VALIDATION_FAILED":
-                    raise InvalidComponent("Invalid component.")
-                if retry_after > 0:
-                    self.logger.log(level="Warning", msg=f"Ratelimited while running {name}, retrying after {retry_after}")
-                    time.sleep(retry_after)
-
-                return None
-            if code == 10003:
-                raise UnknownChannel("Bot doesn't have access to this channel.")
-
-        lim = time.time() + timeout
-        if message_id == "":
-            while time.time() < lim:
-                if nonce in self.ws_cache:
-                    for i in event_type:
-                        if i in self.ws_cache[nonce]:
-                            _message = self.ws_cache[nonce][i]
-                            return Message(_message)
-
-        if not _message and message_id == "":
-            raise NonceTimeout("Did not receive nonce in time.")
-
-        lim = time.time() + timeout
-        while time.time() < lim:
             try:
-                for key, value in self.ws_cache.items():
-                    try:
-                        event = key.split(" ")[1]
-                        if value["id"].strip() == message_id.strip() and event in event_type:
-                            _message = value
-                            return Message(_message)
-                        if (
-                            value["message_reference"]["message_id"].strip() == message_id.strip
-                            and value["type"] == 19
-                            and event in event_type
-                        ):
-                            _message = value
-                            return Message(_message)
-                    except Exception as e:
-                        pass
-            except:
-                pass
-
-        if not _message:
-            raise NonceTimeout("Did not receive nonce in time.")
+                interaction: Optional[Union[Message, bool, None]] = self.wait_for("MESSAGE_CREATE", nonce)
```

Comment:
You don't need to typehint None in an optional union. The entire point of using Optional is signifying that the function may not return anything at all, i.e. None